src/expr.rs

```diff
@@ -854,19 +829,58 @@ fn rewrite_call(context: &RewriteContext,
                 -> Option<String> {
     debug!("rewrite_call, width: {}, offset: {}", width, offset);
 
-    // FIXME using byte lens instead of char lens (and probably all over the place too)
     // 2 is for parens
-    let max_callee_width = try_opt!(width.checked_sub(2));
-    let callee_str = try_opt!(callee.rewrite(context, max_callee_width, offset));
-    debug!("rewrite_call, callee_str: `{}`", callee_str);
-
-    if args.is_empty() {
-        return Some(format!("{}()", callee_str));
+    let mut hi = try_opt!(width.checked_sub(2)) * 2;
+    let mut lo = 0;
+    let mut tries = 0;
+
+    // Binary search for the best split between callee and arguments.
```

Comment:
It actually takes the first reasonable answer it can get. 10 was taken as a safety measure, indeed. 2log(10) should be better. I'll be sure to do the refactor and add documentation.